espnet2/asr/espnet_model.py

```diff
@@ -199,8 +199,8 @@ def encode(
             # 3. Normalization for feature: e.g. Global-CMVN, Utterance-CMVN
             if self.normalize is not None:
                 feats, feats_lengths = self.normalize(feats, feats_lengths)
-                if self.adddiontal_utt_mvn is not None:
-                    feats, feats_lengths = self.adddiontal_utt_mvn(feats, feats_lengths)
+                if self.additional_utt_mvn is not None:
```

Comment:
Utterance CMVN after global CMVN doesn't have no effect of global CMVN mathematically.

Reply:
This is for the MIMO-speech case where single-channel and multi-channel data are mixed in training and the gloval mvn stats is provided from single-channel data.

Reply:
Now enh task mixes single-channel and multi-channel ?

Reply:
It seems no effect even MIMO mathematically? What do you means? 
 > This is for the MIMO-speech case where single-channel and multi-channel data are mixed in training and the gloval mvn stats is provided from single-channel data.

Reply:
Sorry for the confusion. I'll explain about it more. In terms of MIMO, the normalization method is Global_MVN，the stats of which is originally computed from single-channel input. Then for multi-channel input, an additional Utt_MVN is needed for training stability. This is to be consistent with the experiment in ESPnet 1. I guess your point is that the stats for Global_MVN can be computed from all the data (single-/multi-channel) in espnet2. So it is expected to have no effect for Utterance?

Reply:
Just for making sure, `adddiontal_utt_mvn` is calculating the statistics from each utterance, right?

If you'll apply utterance mvn, the global mvn is no effect because it just shift mean and var globally.

```
# These two are equivalent
# Utt CMVN after global CMVN
x_g = (x - mean_global) / std_global
y = (x_g -  x_g.mean()) / x_g.std()

# Only utt CMVN 
y = (x -  x.mean()) / x.std()
```

Reply:
> Now enh task mixes single-channel and multi-channel ?

Yes, we are trying to unify single-channel and multi-channel, single-speaker and multi-speaker in one framework, for better integration with ASR. Otherwise, we need to implement multiple interfaces (`espnet_model`s) for joint training of enh and ASR.

Reply:
Could you answer?

Reply:
In my understanding, you are correct if we first do global **mean-variance** normalization and then utterance **mean-variance** normalization. The global MVN has no effect in that case.

But actually in [our enh_asr task](https://github.com/LiChenda/espnet/blob/2d7a96cf2b8204d89673abfe72f23d8d021c7bdd/espnet2/asr/espnet_enh_asr_model.py#L74) (like MIMO-Speech, not included in this PR), we only do utterance **mean** normalization after global MVN, so the globally normalized variance is still kept.

I might say this operation could have little difference from pure utt-mvn, but maybe @simpleoier has more experience on it.

Reply:
I understood, but I guess in your case, I guess global mean-variance normalization doesn't improve the performance. 
I mean only utterance mean normalization is enough, as far as I did, so I think we should remove adddiontal_utt_mvn.

Reply:
And if you really need `global_var_norm  + utterance_mean_norm`, we should implement it as one of normalization classes instead of `adddiontal_utt_mvn `. 
This is more clear.

Reply:
Yeah, I agree it would be better to have all normalization operations in one single class.

Reply:
Yeah, I have removed it.